src/pages/DoctorDashboard.tsx

```diff
@@ -9,12 +10,156 @@ import {
+import { useAuth } from '../App';
+
 function DoctorDashboard() {
     const navigate = useNavigate()
+    let today = getBeginningOfWeek(new Date());
+    today.setHours(0,0,0,0);
+    const [startDate, onStartDateChange] = useState<Date>(today);
+    const [endDate, onEndDateChange] = useState<Date>(addDays(today, 6));
+    const [reserved, setReserved] = useState<string>('-1');
+    const [page, setPage] = useState<number>(1);
+    const [maxPage, setMaxPage] = useState<number>(10);
+    const radios = [
+        { name: 'All', value: '-1' },
+        { name: 'Free', value: '0' },
+        { name: 'Reserved', value: '1' },
+    ];
+    let auth: AuthContextType = useAuth();
+    const sampleInfo: Visit[] = [
+        {
+            date: new Date("2022-03-29T08:00:00Z"),
+            id: 0,
+            vaccination: null,
+        },
+        {
+            date: new Date("2022-03-30T08:00:00Z"),
+            id: 1,
+            vaccination: {
+                id: 0,
+                status: "Planned",
+                patient: {
+                    id: 0,
+                    firstName: "Adam",
+                    lastName: "Abacki",
+                    pesel: "12345678901",
+                    email: "email@example.com",
+                    address: {
+                        id: 0,
+                        city: "Warszawa",
+                        zipCode: "01-234",
+                        street: "Przykładowa",
+                        houseNumber: 1,
+                        localNumber: null,
+                    }
+                },
+                vaccine: {
+                    id: 0,
+                    name: "Phiser",
+                    disease: "Covid-19",
+                    requiredDoses: 0,
+                },
+            }
+        },
+        {
+            date: new Date("2022-03-29T10:00:00Z"),
+            id: 0,
+            vaccination: null,
+        },
+    ];
+    function getVisits(start: Date | null, finish: Date | null, onlyReserved: string, page: number): Visit[] {
+        getSlots(start, finish, onlyReserved, auth.token, page).then((response) => {
+            if(response.ok)
+                setMaxPage(response.pagination.totalPages);
+                return response.data;
+        }).catch((reason => {
+            switch (reason)
+            {
+                case 401:
+                    console.log("Unauthorised error (invalid or empty Bearer token");
+                    break;
+                case 422:
+                    console.log("Validation error");
+                    break;
+                default:
+                    console.log("Unknown error");
+            }
+        }))
+        return [];
+    }
+    const [Visits, setVisits] = useState<Visit[]>([]);
+    useEffect(() => {
+        setVisits(getVisits(startDate, endDate, reserved, page));
+    },[startDate, endDate, reserved, page])
+    function remove(index: number) {
+        Visits.splice(index, 1);
```

Comment:
Wydaje mi się, że to może nie zadziałać. (Ale nie sprawdzałem). Bo z tego co kojarzę to zawsze jak chcesz zmienić stan np "Visits", to powinieneś używać "setVisits". Czyli np setVisits(i tutaj należy wstawić jakąś funkcję, która ZWRÓCI wszystkie elementy Visits oprócz tego jenego wskazanego).

Reply:
Ewentualnie zawsze można wywołać raz jeszcze getVisits(), aby pobrało wszystkie rekordy raz jeszcze. I myślę, że to może być najlepsze rozwiązanie.